File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/models/JobOpening/ResearchOpening.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PlacementCellBackend.Models
{
    public class ResearchOpening
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; } = string.Empty;

        // Teacher who posted the research opening
        [Required]
        public string TeacherId { get; set; } = string.Empty;

        [ForeignKey("Id")]
        public Teacher? Teacher { get; set; }

        // Research opening details
        public string Title { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public string Department { get; set; } = string.Empty;

        public string Researcharea { get; set; } = string.Empty;

        public string Stipend { get; set; } = string.Empty;

        public string Duration { get; set; } = string.Empty;

        public DateOnly PostedDate { get; set; }

        public DateOnly? DeadLine { get; set; }

        public string Link { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/models/JobOpening/ResearchOpening.java
package com.example.placementicsbackend.models.JobOpening;

import com.example.placementicsbackend.models.Teacher;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "teacherresearchopening")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ResearchOpening {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer recordId;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "Id", referencedColumnName = "Id")
    private Teacher teacher;

    @Column(name = "Title", nullable = false)
    private String title = "";

    @Column(name = "Description", nullable = false)
    private String description = "";

    @Column(name = "Department", nullable = false)
    private String department = "";

    @Column(name = "Researcharea", nullable = false)
    private String researchArea = "";

    @Column(name = "Stipend", nullable = false)
    private String stipend = "";

    @Column(name = "Duration", nullable = false)
    private String duration = "";

    @Column(name = "PostedDate", nullable = false)
    private LocalDate postedDate;

    @Column(name = "DeadLine")
    private LocalDate deadline;

    @Column(name = "Link", nullable = false)
    private String link = "";

    @Column(name = "IsActive", nullable = false)
    private boolean active = true;
}
